c/serverStatusService.c

```diff
@@ -0,0 +1,272 @@
+  int ziipUtil = 0x00000000;
+  uname(&unameRet);
+  gethostname(hostnameBuffer, sizeof(hostnameBuffer));
+  buffer = (char *)safeMalloc(bufferlen, "buffer");
+  memset(buffer, 0, bufferlen);
+  smfFunc = (EXSMFI *)fetch("ERBSMFI");
+  rc = (*smfFunc)(&reqtype,
+               &rectype,
+               &subtype,
+               buffer,
+               &bufferlen,
+               &cpuUtil,
+               &demandPaging,
+               &options,
+               &mvsSrm,
+               &zaapUtil,
+               &ziipUtil);
+  safeFree(buffer, bufferlen);
+  curr_time = time(NULL);
+  c_time_str = ctime(&curr_time);
+  if(c_time_str[strlen(c_time_str) - 1] == '\n'){
+    c_time_str[strlen(c_time_str) - 1] = '\0';
+  }
+  if(rc > 0){
+    respondWithError(response, HTTP_STATUS_BAD_REQUEST, "Unable to fetch from RMF data interface service");
+    return;
+  }
+  jsonPrinter *out = respondWithJsonPrinter(response);
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+  writeHeader(response);
+  jsonStart(out);
+  jsonAddString(out, "timestamp", c_time_str);
+  if(getenv("ZSS_LOG_FILE") != NULL){
+    jsonAddString(out, "logDirectory", getenv("ZSS_LOG_FILE"));
+  } else {
+    jsonAddString(out, "logDirectory", "ZSS_LOG_FILE not defined in environment variables");
+  }
+  jsonAddString(out, "agentName", "zss");
+  jsonAddString(out, "agentVersion", context->productVersion);
+  jsonAddString(out, "arch", unameRet.sysname);
+  jsonAddString(out, "osRelease", unameRet.release);
+  jsonAddString(out, "osVersion", unameRet.version);
+  jsonAddString(out, "hardwareIdentifier", unameRet.machine);
+  jsonAddString(out, "hostname", hostnameBuffer);
+  jsonAddString(out, "nodename", unameRet.nodename);
+  jsonStartObject(out, "userEnvironment");
+  char *env_var = *environ;
+  for (int i = 1; env_var; i++) {
+    char *equalSign = strchr(env_var, '=');
+    if(equalSign == NULL){
+      break;
+    }
+    int nameLen = strlen(env_var) - strlen(equalSign);
+    int nameBufferLen = nameLen + 1;
+    char *name = safeMalloc(nameBufferLen, "env_var name");
+    memcpy(name, env_var, nameLen);
+    name[nameLen] = '\0';
+    jsonAddString(out, name, equalSign+1);
+    safeFree(name, nameBufferLen);
+    env_var = *(environ+i);
+  }
+  jsonEndObject(out);
+  jsonAddInt(out, "demandPagingRate", demandPaging);
+  jsonAddInt(out, "stdCP_CPU_Util", cpuUtil);
+  jsonAddInt(out, "stdCP_MVS_SRM_CPU_Util", mvsSrm);
+  jsonAddInt(out, "ZAAP_CPU_Util", zaapUtil);
+  jsonAddInt(out, "ZIIP_CPU_Util", ziipUtil);
+  jsonAddInt(out, "PID", getpid());
+  jsonAddInt(out, "PPID", getppid());
+  jsonEnd(out);
+  finishResponse(response);
+}
+
+static int serveStatus(HttpService *service, HttpResponse *response) {
+  HttpRequest *request = response->request;
+  ServerAgentContext *context = service->userPointer;
+  //This service is conditional on RBAC being enabled because it is a sensitive URL that only RBAC authorized users should be able to access
```

Comment:
This line is more than what's readable without scrolling on Github. If you go back and make changes, you'd might as well fix this line length.